src/jdk.jshell/share/classes/jdk/internal/jshell/tool/JShellTool.java

```diff
@@ -1687,7 +1686,8 @@ public boolean matchesType() {
                                     return false;
                                 }
                             })
-                            .collect(toList());
+                            .map(e -> (Suggestion) e)
```

Comment:
I guess I'd get rid of the cast here as well. Otherwise looks good!

Reply:
Ah yes, missed that one. Done!